### PR TITLE
[10.0[MIG]exchangelib

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
It seems to work in our tests for simple calendar events and contacts with single addresses. Pending to tests for more complex cases.

@damdam-s

### CODE REVIEW

File: connector_exchange/data/cron.xml [hunk 1]
```diff
@@ -1,13 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
-  <data noupdate="1">
+<odoo>
 
     <record forcecreate="True" id="ir_cron_sync_import_contact_partner" model="ir.cron">
       <field name="name">Connector Exchange - Import Contacts</field>
-      <field eval="False" name="active"/>
```
Comment (damdam-s) [inline connector_exchange/data/cron.xml:7]: I would have kept the cron inactive by default and activate it only when needed

File: connector_exchange/models/calendar_event/exporter.py [hunk 2]
```diff
@@ -104,138 +96,142 @@ def fill_free_busy_status(self, calendar):
...
         In Exchange, only one reminder can be set.
         So Odoo side, only the first reminder will be exported to Exchange.
         """
         alarms = self.binding.alarm_ids
         if alarms:
             alarm = alarms[0]
-            calendar.is_reminder_set.set(True)
-            calendar.reminder_due_by.set(convert_to_exchange(
+            event.is_reminder_set = True
+            event.reminder_due_by = convert_to_exchange(
```
Comment (damdam-s) [inline connector_exchange/models/calendar_event/exporter.py:112]: maybe we should use the `parse_date` method defined below

File: connector_exchange/models/calendar_event/exporter.py [hunk 3]
```diff
@@ -104,138 +96,142 @@ def fill_free_busy_status(self, calendar):
...
-                    attendee.email, calendar)):
-                att = Attendee()
-                att.mailbox.name.set(attendee.common_name)
-                att.mailbox.email_address.set(attendee.email)
-                att.response_type.set(
-                    STATES_MAPPING.get(attendee.state,
-                                       ResponseTypeType.Unknown)
+                    attendee.email, event)):
+                att = Attendee(
+                    mailbox=Mailbox(email_address='attendee.email',
```
Comment (damdam-s) [inline connector_exchange/models/calendar_event/exporter.py]: remove quotes around `attendee.email`

**Review by damdam-s — COMMENTED** [review]
just a quick code review. starting tests right now :smile:

File: connector_exchange/data/cron.xml (hunk 1 above)
Comment (AaronHForgeFlow) [reply connector_exchange/data/cron.xml:7]: Yes, that was done just for easy testing. I'll put that back. Thanks

File: connector_exchange/models/exchange_backend/common.py [hunk 4]
```diff
@@ -42,15 +33,8 @@ def select_versions(self):
         """
         return [('exchange_2010', 'Exchange 2010')]
 
+    # todo remove version, no needed in exchangelib
     version = fields.Selection(selection='select_versions', required=True)
-    certificate_location = fields.Char(
-        required=True,
-        default="/opt/exchange_ws/certificates/")
-    location = fields.Char(
```
Comment (damdam-s) [inline connector_exchange/models/exchange_backend/common.py:49]: I must be blind but how/where do you configure the URL to find the exchange server?

Comment (damdam-s) [reply connector_exchange/models/exchange_backend/common.py:49]: I cannot see something like
```
config = Configuration(server='mail.example.com', credentials=credentials)
account = Account(primary_smtp_address='john@example.com', config=config,
                  autodiscover=False, access_type=IMPERSONATION)
```
found in the documentation of the exchangelib

Comment (AaronHForgeFlow) [reply connector_exchange/models/exchange_backend/common.py:49]: I seems to me it is taken from the domain of the email of the impersonation account. I will recheck, but I was able to connect without providing the URL and the certificate.

File: connector_exchange/models/calendar_event/exporter.py (hunk 2 above)
Comment (AaronHForgeFlow) [reply connector_exchange/models/calendar_event/exporter.py:112]: Good point!

File: connector_exchange/models/exchange_backend/common.py (hunk 4 above)
Comment (damdam-s) [reply connector_exchange/models/exchange_backend/common.py:49]: ok. my customer does not use external email address to use the connector. I will add an option to disable autodiscover feature and provide a config object instead.

Comment (AaronHForgeFlow) [reply connector_exchange/models/exchange_backend/common.py:49]: You are right. Maybe we can add a boolean field to indicate autodiscover or not.

Comment (damdam-s) [reply connector_exchange/models/exchange_backend/common.py:49]: that's what I was thinking about. I will make a PR against yours :smile:

**Comment by damdam-s** [comment]
@aheficent I managed to connect via specific URL. 

Now, when exporting a calendar event, I have a 6h difference between Odoo and Exchange. Any idea on this?

**Comment by AaronHForgeFlow** [comment]
Good. I think the timezone is hardcoded to East Coast in USA. My fault. I'll fix that.

**Pull request merged**